plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/rest/TrinoIcebergRestCatalogFactory.java

```diff
@@ -108,6 +110,17 @@ public synchronized TrinoCatalog create(ConnectorIdentity identity)
             icebergCatalog = icebergCatalogInstance;
```

Comment:
Github won't let me put a comment on the right line but on line 92 we add security properties to the `properties` map, can we remove that now?

Reply:
oh no its still required and used during initialization but the newly added properties are having effect during token refresh. Its confusing on why we need to provide same properties twice.

Ive expressed the same concern here https://github.com/trinodb/trino/pull/23060#issue-2470331175
>Im not following on why Map<String, String> credentials is explicitly provided and not fetched from AuthSession parent in there, cause AuthSession has AuthConfig which has token/credential.